core/engine.go

```diff
@@ -232,7 +233,50 @@ type Injectable interface {
 	Config() interface{}
 }
 
-func getEngineName(engine Engine) string {
+// HealthCheckable is the interface for engines that can perform health checks, which result can be reported to monitoring tooling.
+type HealthCheckable interface {
+	// CheckHealth performs health checks and returns the result. The caller may cancel the operation if it takes too long,
+	// so it should observe the context and abort on cancellation.
+	CheckHealth(ctx context.Context) map[string]HealthCheckResult
+}
+
+// HealthStatus defines the result status of a health check.
+type HealthStatus string
+
+// Severity returns an integer indicating the seriousness of a startup.
+// 0 means all is OK, higher is worse.
+func (h HealthStatus) Severity() int {
+	switch h {
+	case HealthStatusDown:
+		return 2
+	case HealthStatusUnknown:
+		return 1
+	default:
+		fallthrough
+	case HealthStatusUp:
+		return 0
+	}
+}
+
+const (
+	// HealthStatusUp indicates the health check succeeded
+	HealthStatusUp HealthStatus = "UP"
+	// HealthStatusDown indicates the health check failed
+	HealthStatusDown HealthStatus = "DOWN"
+	// HealthStatusUnknown indicates the status of the health check is unknown
+	HealthStatusUnknown HealthStatus = "UNKNOWN"
+)
+
+// HealthCheckResult is the result of a health check.
+type HealthCheckResult struct {
+	// Status contains the status of the health check.
+	Status HealthStatus `json:"status"`
+	// Details contains optional details of the health check, can be nil.
+	Details interface{} `json:"details,omitempty"`
+}
```

Comment:
```suggestion
// Health is the result of a health check.
type Health struct {
	// Status contains the status of the health check.
	Status HealthStatus `json:"status"`
	// Details contains optional details of the health check, can be nil.
	Details interface{} `json:"details,omitempty"`
}
```

Just `Health.{Status,Details}` looks very readable to me